### DISCUSSION TITLE
New to developing - Where are all the mod.manifest files?

### DISCUSSION THREAD

**Discussion by chrcwalk** [TIBCOSoftware/spotfire-mods#118] [Q&A] [answered]
Hey all, love the community and collaborative modding efforts here - it's helped me a ton in visualizing data for others. 

I've downloaded some of the sample mods from the repository by the user objerke as well as some others, but a large portion of the mods do not have a mod.manifest file in the 'src' folder. This is important because i don't really have access to node.js, meaning i can't even run an nmp install command. I was curious as to why some of these mods have a mod.manifest file in the 'src' folder (which is the only way for me to pull it into Tibco), and others don't. I'd love to dig into developing and modding more - my ultimate goal is to add in custom coloring to the gauge chart mod so that the gauges will reflect different colors based upon a certain value being reached for the gauge itself.

Thanks for all your help and all that everyone does here!!

**Comment by jagattibco** [comment, marked as answer]
chrcwalk,

The [example mods](https://github.com/TIBCOSoftware/spotfire-mods/tree/master/examples) named "ts-*" are written in TypeScript. These cannot be imported and used directly by Spotfire but require a build step where the TS code is transpiled to JS and bundled up with static resources used by the mod. We have chosen to structure the TS mods so that the static resources are all placed in the "static" folder keeping them separate from the TS code in "src". We just think this is a nice way of structuring a TS mod project.
As you have noticed, the mod-manifest.json is considered a static resource and placed in the "static" folder rather than in "src".

You say that you don't have access to node.js and npm. I suppose that means you do not have access to TypeScript either?
If that is the case I suppose your only option is to treat the TS examples as inspiration and manually rewrite them in JS and structure the code and folders to meet your constraints.
